### PR TITLE
MINOR: Add “npm run add-git-hooks” action.

### PR DESCRIPTION
**Pull request by sminnee**
This commit add an "npm run add-git-hooks" command that will configure
an environment's git hooks and config file to support the development
conventions that we set out.

It's based on recommendations from this post:
http://blog.andrewray.me/dealing-with-compiled-files-in-git/

More git hooks can be added to .githooks

To do:
- [ ] Test what happens when the merge failures happen without the git config
- [ ] Test what happens when you a conflicting non-rebase merge happens
- [ ] Add docs to `en/05_Contributing/01_Code.md#npm`
- [ ] Add checks for SCSS files

### CODE REVIEW

**Comment by hafriedlander** [comment]
With the change to .gitattributes being immediate, but the addition of the merge strategy only happening after "npm run add-git-hooks", have you checked what happens if add-git-hooks isn't run? Does it just error out?

(check-built-files.sh is also pretty lossy - won't pick up scss changes for instance - but probably good enough as a start)

**Comment by hafriedlander** [comment]
Something else I wanted to test: what happens if you use merge instead of rewrite and you get a clash, does "post-rewrite" still get called?

File: package.json [hunk 1]
```diff
@@ -18,7 +18,8 @@
     "test": "NODE_PATH=\"./javascript/src:./admin/javascript/src\" jest",
     "coverage": "NODE_PATH=\"./javascript/src:./admin/javascript/src\" jest --coverage",
     "thirdparty": "gulp thirdparty",
-    "lint": "eslint gulpfile.js & eslint javascript/src & eslint admin/javascript/src"
+    "lint": "eslint gulpfile.js & eslint javascript/src & eslint admin/javascript/src",
+    "add-git-hooks": "./node_modules/.bin/git-hooks --install; git config merge.ours.driver true"
```
Comment (flashbackzoo) [inline package.json]: You should be able to just do `git-hooks --install; git config merge.ours.driver true`. See https://docs.npmjs.com/misc/scripts#path

**Comment by flashbackzoo** [comment]
We've been documenting npm scripts https://github.com/silverstripe/silverstripe-framework/blob/master/docs/en/05_Contributing/01_Code.md#npm can you please add the new script?

**Comment by sminnee** [comment]
To do:
 - [ ] Test what happens when the merge failures happen without the git config
 - [ ] Test what happens when you a conflicting non-rebase merge happens
 - [ ] Add docs to `en/05_Contributing/01_Code.md#npm`
 - [ ] Add checks for SCSS files

**Comment by sminnee** [comment]
Abandoning this as stale.

**Pull request closed without merging**